### ISSUE TITLE
2nd Easee charger gives TypeError on custom command for dynamicCurrent setting

### ISSUE THREAD

**Issue by Deepfreeze10** [runnane/node-red-contrib-easee#25]
I've got 2 Easee chargers and use the node-red-contrib-easee (v0.6.2) with node-red (v3.1.5).
Charger 1 can handle custom commands fine for dynamicCurrent setting. But when sending the message to the second charger, I get a TypeError: Cannot convert undefined or null to object. 

Initially my message looks like this:
msg.payload = {"path":"/sites/###943/circuits/690###/dynamicCurrent","body":{"phase1":0,"phase2":16,"phase3":0}}

But I've also tried with the "method":"POST" option:
msg.payload = {"method":"POST","path":"/sites/###943/circuits/690###/dynamicCurrent","body":{"phase1":0,"phase2":16,"phase3":0}}

Any idea on how to get my second charger functioning with custom commands? The other commands like login, refresh_token, charger_state work fine on both first (siteid: ###944) and second charger (siteid: ###943).

**Comment by runnane** [comment]
Hi,

Are you able to post the full stack trace of the error? I might have to add some error catching if there are some limitations. Do you have the same access level on both chargers? I have access to two myself, and one of them i am only a user and I'm not allowed to change current settings. It is currently offline so I cannot test/reproduse, but I will try when available.

Jon

**Comment by Deepfreeze10** [comment]
Hi Jon,

My trace log shows the following:
Error: REST Command failed (404: Not Found) {"errorCode":230,"errorCodeName":"CircuitDoesNotBelongToSite","type":null,"title":"Circuit does not belong to site","status":404,"detail":"[Empty in production]","instance":null,"extensions":{}} 
at EaseeConfiguration.node.doAuthRestCall (/data/home/nodered/.node-red/node_modules/@runnane/node-red-contrib-easee/easee-client/easee-configuration.js:123:21) 
at processTicksAndRejections (node:internal/process/task_queues:95:5)

So I checked if my circuit id is correct, using the easee site command (see below). I see the same for both charger id's. So I have a single site, a single circuit and 2 backplates / chargers on a single circuit. My left charger is the master / leader. Could your code work with this or is there a possibility to create either 2 sites or 2 circuits with each a single charger?

{
  "uuid": "ad...",
  "id": ###944,
  "siteKey": "T4JG-####",
  "name": "PC Laadlocatie (Links)",
  "accessMode": 2,
  "levelOfAccess": 1,
  "address": {

    },
    "latitude": null,
    "longitude": null,
    "altitude": null
  },
  "contactInfo": {
    "installerName": null,
    "installerPhoneNumber": null,
    "ownerName": "##",
    "ownerPhoneNumber": "+##",
    "company": null
  },
  "costModel": "fixedprice",
  "costModelVariant": null,
  "costPerKWh": 0.23,
  "costPerKwhExcludeVat": 0.23,
  "currencyId": "EUR",
  "siteType": 1,
  "siteCategory": 0,
  "ratedCurrent": 25,
  "vat": 0,
  "partnerId": 20,
  "installerId": null,
  "installerUserId": null,
  "supportId": null,
  "installerAlias": "",
  "useDynamicMaster": false,
  "circuits": [
    {
      "id": 690###,
      "uuid": "b5a369a3##",
      "siteId": ###944,
      "circuitPanelId": 1,
      "panelName": "EV groep",
      "ratedCurrent": 16,
      "fuse": null,
      "chargers": [
        {
          "id": "EH2G####",
          "name": "EV lader Pulsar (Rechts)",
          "color": 1,
          "createdOn": "2022-12-23T07:30:06.475256",
          "updatedOn": "2023-12-30T08:55:37.913407",
          "backPlate": {
            "id": "8172B##",
            "masterBackPlateId": "8072B##",
            "name": "EV laadlocatie (Rechts)",
            "backplateType": 0,
            "features": []
          },
          "levelOfAccess": null,
          "productCode": 2,
          "userRole": 1,
          "isTemporary": false
        },
        {
          "id": "EH6Q####",
          "name": "EV lader Leaf (Links)",
          "color": 1,
          "createdOn": "2022-12-23T05:50:02.938067",
          "updatedOn": "2023-12-30T08:53:35.277587",
          "backPlate": {
            "id": "8072B5##",
            "masterBackPlateId": "8072B5##",
            "name": "EV laadlocatie (Links)",
            "backplateType": 0,
            "features": []
          },
          "levelOfAccess": null,
          "productCode": 2,
          "userRole": 1,
          "isTemporary": false
        }
      ],
      "masterBackplate": null,
      "useDynamicMaster": false,
      "parentCircuitId": null
    }
  ],
  "equalizers": [],
  "createdOn": "2023-12-30T08:34:02.890327",
  "updatedOn": "2024-11-30T16:01:14.911397",
  "userRole": 1,
  "allowedSiteActions": [
    "AllowToConfigureLevelOfAccess"
  ],
  "support": null,
  "regulations": []
}

**Comment by Deepfreeze10** [comment]
I've also tested if I would get the error on the easee api page:
https://developer.easee.com/reference/post_api-sites-siteid-circuits-circuitid-dynamiccurrent

I've got the same error. So I used the installer app to add a circuit on the second charger. Now each site has it's own circuit and each charger is master within it's own circuit.

Now the payload for setting the dynamic currents works fine!
msg.payload = {"path":"/sites/###943/circuits/817###/dynamicCurrent","body":{"phase1":0,"phase2":16,"phase3":0}}

Thanks for the hint to look at the trace log file!
